tests/data_context/store/test_expectations_store.py

```diff
@@ -310,8 +310,262 @@ def test_get_key_in_non_cloud_mode(empty_data_context):
 @pytest.mark.unit
 def test_get_key_in_cloud_mode(empty_data_context_in_cloud_mode):
     cloud_data_context = empty_data_context_in_cloud_mode
-    project_manager.set_project(cloud_data_context)
     name = "test-name"
     suite = ExpectationSuite(expectation_suite_name=name)
     key = cloud_data_context.expectations_store.get_key(suite)
     assert isinstance(key, GXCloudIdentifier)
+
+
+@pytest.mark.cloud
+def test_add_expectation_success_cloud_backend(empty_cloud_data_context):
+    # Arrange
+    context = empty_cloud_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert expectation.column == added_expectation.column
+    assert expectation.value_set == added_expectation.value_set
+    assert expectation.result_format == added_expectation.result_format
+
+
+@pytest.mark.filesystem
+def test_add_expectation_success_file_backend(empty_data_context):
+    # Arrange
+    context = empty_data_context
+    store = context.expectations_store
+    suite_name = "test-suite"
+    suite = context.add_expectation_suite(suite_name)
+    key = store.get_key(suite)
+    expectation = ExpectColumnValuesToBeInSet(
+        column="a",
+        value_set=[1, 2, 3],
+        result_format="BASIC",
+    )
+
+    # Act
+    store.add_expectation(suite=suite, expectation=expectation)
+
+    # Assert
+    updated_suite_dict = store.get(key=key)
+    updated_suite = ExpectationSuite(**updated_suite_dict)
+    added_expectation = updated_suite.expectations[0]
+    assert UUID(added_expectation.id)
+    assert expectation.column == added_expectation.column
+    assert expectation.value_set == added_expectation.value_set
+    assert expectation.result_format == added_expectation.result_format
+
+
+@pytest.mark.filesystem
+def test_add_expectation_disregards_provided_id(empty_data_context):
```

Comment:
I wonder if this is the behavior we want or if we want to raise when an id is already present. I guess the good with this is we can `copy` expectations easily?

Reply:
I wrote a long blurb in slack. My current thought is to `raise` here. I think `copy` should likely not copy `ids` since the use case is to move the expectation to a new suite? We could parameterize `copy` so one can copy ids if they want to by explicit asking for it.

Reply:
`ExpectationSuite.add()` raises if an `Expectation` has an ID, and the `ExpectationsStore` just overwrites. If the inconsistent behavior is weird I can update. My reasoning was that the user should always get an error from the `ExpectationSuite`, and I didn't want to need to try to catch an error from the store as well.

Reply:
i've implemented `copy` in #9155, and it doesn't copy IDs - if its okay, it would be easiest to leave that change for that PR